src/tests/ftest/container/container_async.py

```diff
@@ -57,52 +47,27 @@ def cb_func(event):
     GLOB_RC = event.event.ev_error
     GLOB_SIGNAL.set()
 
-class ContainerAsync(Test):
+class ContainerAsync(TestWithServers):
     """
     Tests DAOS pool connect permissions (non existing pool handle, bad uuid)
     and close.
+
+    :avocado: recursive
     """
 
     def __init__(self, *args, **kwargs):
         super(ContainerAsync, self).__init__(*args, **kwargs)
         self.container1 = None
         self.container2 = None
-
-    def setUp(self):
-        self.agent_sessions = None
-        self.hostlist = None
         self.pool = None
 
-        # get paths from the build_vars generated by build
-        with open('../../../.build_vars.json') as build_file:
-            build_paths = json.load(build_file)
-        self.basepath = os.path.normpath(build_paths['PREFIX'] + "/../")
-
-        self.server_group = self.params.get("name", '/server_config/',
-                                            'daos_server')
-
-        # setup the DAOS python API
-        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
-        self.pool = None
-
-        self.hostlist = self.params.get("test_machines", '/run/hosts/*')
-        self.hostfile = write_host_file.write_host_file(self.hostlist,
-                                                        self.workdir)
-        print("Host file is: {}".format(self.hostfile))
-
-        self.agent_sessions = AgentUtils.run_agent(self.basepath, self.hostlist)
-        server_utils.run_server(self.hostfile, self.server_group, self.basepath)
-        time.sleep(10)
-
     def tearDown(self):
         try:
             if self.pool is not None and self.pool.attached:
                 self.pool.destroy(1)
         finally:
-            if self.agent_sessions:
-                AgentUtils.stop_agent(self.hostlist, self.agent_sessions)
             time.sleep(5)
-            server_utils.stop_server(hosts=self.hostlist)
+            super(ContainerAsync, self).tearDown()
```

Comment:
This new behavior does not match the original.  In order to replicate the original tear down we need an optional `sleep()` between the `stop_agent()` and `stop_server()` calls in `TestWithServers.tearDown()`.  What do you think about a new class attribute, like `self.tear_down_delay=0`, that in this case would be set/overridden to 5 in `__init__()`?

Reply:
Yes, I don't believe there is any particular requirement for this `sleep(5)` and it was probably added as a guess that it might be needed.  I don't think it occurs in any other instances of agent stopping followed by server stopping and if it actually is needed it should be added more wholesomely.  But I would advocate not going down that path until it is clear that it is necessary.

Moreover, if there is some kind of barrier needed between the agent stopping and the servers stopping then testing (but again, I don't believe there is) that specific barrier would be the correct solution vs. adding random sleeps.

Reply:
In talking to Stephen, maybe we just need to remove the `sleep()`.  Is it really needed?

Reply:
Yes, that is my suspicion also.  It's superfluous.

Reply:
ok